libs/feeds_processing/src/adfs_gen_calldata.rs

```diff
@@ -48,7 +47,11 @@ pub async fn adfs_serialize_updates(
     net: &str,
     feed_updates: &BatchedAggegratesToSend,
     feeds_metrics: Option<Arc<RwLock<FeedsMetrics>>>,
-    feeds_config: Arc<RwLock<HashMap<u32, FeedConfig>>>,
+    feeds_config: HashMap<u32, FeedStrideAndDecimals>,
+    feeds_rounds: &mut HashMap<u32, u64>, /* The rounds table for the relevant feeds. If the feeds_metrics are provided,
+                                          this map will be filled with the update count for each feed from it. If the
+                                          feeds_metrics is None, feeds_rounds will be used as the source of the updates
+                                          count. */
```

Comment:
In terms of implementation this is good. In terms of design, I'm not so sure.

We now have an argument, that depending on another argument is sometimes an input parameter and sometimes an output parameter.

This is a rather complicated design in my opinion. I don't have a better suggestion, though.

Reply:
I agree that it is a bit complicated, but we need to be able to call this function both from within the sequencer (which manages the feed round buffer offsets) and from within the reporter (which receives those offsets from the sequencer and uses the provided values as input).